lib/reply.js

```diff
@@ -462,21 +462,17 @@ function sendStream (payload, res, reply) {
   })
 
   eos(res, function (err) {
-    if (err != null) {
-      if (sourceOpen) {
-        if (res.headersSent) {
-          if (!errorLogged) {
-            errorLogged = true
-            logStreamError(reply.log, err, res)
-          }
-        }
-        if (payload.destroy) {
-          payload.destroy()
-        } else if (typeof payload.close === 'function') {
-          payload.close(noop)
-        } else if (typeof payload.abort === 'function') {
-          payload.abort()
-        }
+    if (sourceOpen) {
+      if (err != null && res.headersSent && !errorLogged) {
+        errorLogged = true
+        logStreamError(reply.log, err, res)
+      }
+      if (payload.destroy) {
```

Comment:
why note `typeof payload.destroy === 'function'` ?

Reply:
Good question, I can change it if you want (maybe in another pull request because I will need to add a test for it)

Reply:
@mcollina would have the answer for it.

Reply:
Using typeof would be better, yes but not necessary needed.